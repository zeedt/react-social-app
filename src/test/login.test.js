import { mount, configure } from 'enzyme';
import Login from '../login/login'
import renderer from 'react-test-renderer'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom';
import { withRouter } from 'react-router-dom';
import { Button } from 'react-bootstrap';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const LoginWrappedComponent = withRouter(Login)
describe('Login page Test', () => {
    it('should be able to render login page successfully', () => {
        const component = renderer.create(<Router><LoginWrappedComponent /></Router>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot()
    });

    it('should have login button', () => {
        const wrapper = mount(<Router><LoginWrappedComponent /></Router>);
        expect(wrapper.find(Button).length).toEqual(1);
    });

    it('should have text that ask user if they do not already have an account ', () => {
        const wrapper = mount(<Router><LoginWrappedComponent /></Router>);
        expect(wrapper.find('.userHelpDiv').text()).toEqual(expect.stringContaining('Don\'t have an account?'));
    });

});
