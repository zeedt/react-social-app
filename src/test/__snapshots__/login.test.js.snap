// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Test should be able to render login page successfully 1`] = `
<div
  className="wrapper"
>
  <div
    className="login-container justify-content-center"
  >
    <div
      className="brand_logo_container"
    >
      <img
        alt="Logo"
        className="brand_logo"
        src="/image/app-logo-image"
      />
    </div>
    <div
      className="login-box"
    >
      <div
        className=""
      >
        <form
          className="col-10 offset-1 loginForm"
          onSubmit={[Function]}
        >
          <div
            className="input-group mb-3"
          >
            <div
              className="errorDiv col-12"
            >
              
            </div>
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text ficon"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user fa-w-14 "
                  color="white"
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </span>
            </div>
            <input
              className="form-control input_user"
              name="username"
              onChange={[Function]}
              placeholder="username"
              type="text"
            />
          </div>
          <div
            className="input-group mb-3"
          >
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text ficon"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-key fa-w-16 "
                  color="white"
                  data-icon="key"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M512 176.001C512 273.203 433.202 352 336 352c-11.22 0-22.19-1.062-32.827-3.069l-24.012 27.014A23.999 23.999 0 0 1 261.223 384H224v40c0 13.255-10.745 24-24 24h-40v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-78.059c0-6.365 2.529-12.47 7.029-16.971l161.802-161.802C163.108 213.814 160 195.271 160 176 160 78.798 238.797.001 335.999 0 433.488-.001 512 78.511 512 176.001zM336 128c0 26.51 21.49 48 48 48s48-21.49 48-48-21.49-48-48-48-48 21.49-48 48z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </span>
            </div>
            <input
              className="form-control input_user"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
            />
          </div>
          <div
            className="input-group"
          >
            <button
              className="form-control btn btn-danger md-col-12 pull-right loginButton btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                Login
              </span>
            </button>
          </div>
        </form>
        <div
          className="userHelpDiv"
        >
          Don't have an account? 
          <a
            href="/signup"
            onClick={[Function]}
          >
            Sign Up
          </a>
           
          <p />
          <a
            href="/signup"
            onClick={[Function]}
          >
            Forgot your password?
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
